### PR TITLE
Improve name generator for Sub::Meta

### PR DESCRIPTION
**Pull request by kfly8**
Since Sub::Meta 0.09, the Sub::Meta has the display method, so improved name_generator.

### CODE REVIEW

File: t/10-unit/Factory/name_generator.t [hunk 1]
```diff
@@ -57,17 +57,13 @@ describe 'Use default name generator' => sub {
...
-        parameters => Sub::Meta::Parameters->new(
-          args   => [],
-          slurpy => 1,
-        ),
-        returns => Sub::Meta::Returns->new(),
+        args => [Int, Int],
+        returns => Int,
       );
-      is $factory->name_generator->($factory->name, $meta), $factory->name . "[$meta]";
+      is $factory->name_generator->($factory->name, $meta), $factory->name . "[sub(Int, Int) => Int]";
```
Comment (ybrliiu) [inline t/10-unit/Factory/name_generator.t]: I would like it to be easier to see that the type parameter is an instance of Sub::Meta.

Comment (ybrliiu) [reply t/10-unit/Factory/name_generator.t]: How about this?
```
[Sub::Meta: sub(Int, Int) => Int]
```

Comment (kfly8) [reply t/10-unit/Factory/name_generator.t]: `:` and `::` were confusing.
How about `[Sub::Meta=sub(Int, Int) => Int]`?

Comment (ybrliiu) [reply t/10-unit/Factory/name_generator.t]: Good.

**Pull request merged**